src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.BaseNameProvider.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols
+{
+    internal partial class SymbolTreeInfo
+    {
+        /// <summary>
+        /// Used to produce the simple-full-name components of a type from metadata.
+        /// The name is 'simple' in that it does not contain things like backticks,
+        /// generic arguments, or nested type + separators.  Instead just hte name
+        /// of the type, any containing types, and the component parts of its namespace
+        /// are added.  For example, for the type "X.Y.O`1.I`2, we will produce [X, Y, O, I]
+        /// 
+        /// </summary>
+        private class BaseNameProvider : ISignatureTypeProvider<object>
```

Comment:
Good idea.  @tmat let me know if you think i've implemented this properly, of if there's an easier way to do things.  I did take a look inside SignatureDecoder, and i think this is done correctly.

Essently, given a TypeSpecification for a basetype/interface i want to get the simple name parts of the name.  i.e. for ```NS1.NS2.Outer<X,Y>.Inner<Whatever>``` i want to just get ```[NS1, NS2, Outer, Inner]```, for TypeDefs and TypeRefs this is simple.  For TypeSpecs it was definitely more involved.